src/rust/mqtt-publisher/src/main.rs

```diff
@@ -0,0 +1,230 @@
+        "cancel_order",
+        "fill",
+    ];
+
+    listener.listen_all(channels).await?;
+
+    tokio::task::spawn(async move {
+        loop {
+            eventloop.poll().await.unwrap();
+        }
+    });
+
+    loop {
+        let notification = listener.recv().await?;
+        match notification.channel() {
+            "place_limit_order" => {
+                let data: PlaceLimitOrderNotif = serde_json::from_str(notification.payload())?;
+                client
+                    .publish(
+                        format!(
+                            "place_limit_order/{}/{}/{}/{}",
+                            data.market_id, data.user, data.custodian_id, data.integrator
+                        ),
+                        QoS::AtLeastOnce,
+                        false,
+                        serde_json::to_string(&data)?,
+                    )
+                    .await?;
+            }
+            "place_market_order" => {
+                let data: PlaceMarketOrderNotif = serde_json::from_str(notification.payload())?;
+                client
+                    .publish(
+                        format!(
+                            "place_market_order/{}/{}/{}/{}",
+                            data.market_id, data.user, data.custodian_id, data.integrator
+                        ),
+                        QoS::AtLeastOnce,
+                        false,
+                        serde_json::to_string(&data)?,
+                    )
+                    .await?;
+            }
+            "place_swap_order" => {
+                let data: PlaceSwapOrderNotif = serde_json::from_str(notification.payload())?;
+                client
+                    .publish(
+                        format!(
+                            "place_swap_order/{}/{}/{}",
+                            data.market_id, data.integrator, data.signing_account
+                        ),
+                        QoS::AtLeastOnce,
+                        false,
+                        serde_json::to_string(&data)?,
+                    )
+                    .await?;
+            }
+            "change_order_size" => {
+                let data: ChangeOrderSizeNotif = serde_json::from_str(notification.payload())?;
+                client
+                    .publish(
+                        format!(
+                            "change_order_size/{}/{}/{}",
+                            data.market_id, data.user, data.custodian_id
+                        ),
+                        QoS::AtLeastOnce,
+                        false,
+                        serde_json::to_string(&data)?,
+                    )
+                    .await?;
+            }
+            "cancel_order" => {
+                let data: CancelOrderNotif = serde_json::from_str(notification.payload())?;
+                client
+                    .publish(
+                        format!(
+                            "cancel_order_notif/{}/{}/{}",
```

Comment:
Why is it that some end in `_notif` but some don't?

Reply:
Because my IQ is below freezing temperatures